api/v1/helpers/verbs/putUtils.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * api/v1/helpers/verbs/putUtils.js
+ */
+'use strict'; // eslint-disable-line strict
+
+const u = require('./utils');
+const publisher = u.publisher;
+const event = u.realtimeEvents;
+const httpStatus = require('../../constants').httpStatus;
+const redisCache = require('../../../../cache/redisCache').client.cache;
+
+/**
+ * Generic helper to update a resource.
+ * If no value was provided for an field, clears that field by setting its
+ * value to null (or false for boolean fields).
+ *
+ * @param {Object} req From Express
+ * @param {Object} props Model on which the update operation is performed.
+ * @param {Array} puttableFields From req
+ * @param {Promise} The PUT promise
+ */
+function getPutPromise(req, props, puttableFields) {
+  const toPut = req.swagger.params.queryBody.value;
+  return u.findByKey(
+    props, req.swagger.params
+  )
+  .then((o) => u.isWritable(req, o))
+  .then((o) => {
+    const keys = Object.keys(puttableFields);
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i];
+      if (toPut[key] === undefined) {
+        let nullish = null;
+        if (puttableFields[key].type === 'boolean') {
+          nullish = false;
+        } else if (puttableFields[key].enum) {
+          nullish = puttableFields[key].default;
+        }
+
+        o.set(key, nullish);
+
+        // take nullified fields out of changed fields
+        o.changed(key, false);
+      } else {
+        /**
```

Comment:
only one asterisk here

Reply:
a lot of code here looks the same as patch... any way to call shared fns from either entry point?